File: more-zeroes/moreZeroes.js
const moreZeroes = (string) => {

    const binaries = [];
    string.split('').forEach((letter) => {
        binaries.push(letter.charCodeAt(0).toString(2));
    });
    
    const indices = [];
    binaries.forEach((element, index) => {
        let zeroes = 0;
        for(let i = 0; i < element.length; i++) {
            if(Number(element[i]) === 0) zeroes++;
            if(zeroes >= 4) {
                indices.push(index);
                break;
            };
        };
    });

    const output = [];
    for(let j = 0; j < string.length; j++) {
        if(output.includes(string.charAt(j))) continue;
        if(indices.includes(j) === true) output.push(string.charAt(j));
    };

    return output;
};

module.exports = { moreZeroes };


//https://www.codewars.com/kata/5d41e16d8bad42002208fe1a/train/javascript

File: python-venv/more-zeroes/more_zeroes.py
def more_zeroes(string):

    binaries = []
    indices = []
    characters = [letter for letter in string]

    for i in range(len(characters)):
        binaries.append(bin(ord(characters[i]))[2:len(bin(ord(characters[i])))])

    for j in range(len(binaries)):
        if binaries[j].count('0') >= 4:
            indices.append(j)

    split = list(string)
    output = []

    for k in range(len(split)):
        if split[k] in output:
            continue
        if k in indices:
            output.append(split[k])
            
    return output

#https://www.codewars.com/kata/5d41e16d8bad42002208fe1a/train/python